src/test/java/com/example/ludogoriesoft/lukeriaerpapi/services/CartonServiceTest.java

```diff
@@ -1,144 +1,270 @@
-        existingCarton.setId(cartonId);
-        when(cartonRepository.findById(cartonId)).thenReturn(Optional.of(existingCarton));
-        cartonService.deleteCarton(cartonId);
-        verify(cartonRepository).findById(cartonId);
-        verify(cartonRepository).delete(existingCarton);
+        existingCarton.setId(1L);
+        existingCarton.setName("Existing Carton");
+        existingCarton.setSize("Medium");
+        existingCarton.setAvailableQuantity(5);
+        existingCarton.setPrice(50.0);
+
+        when(cartonRepository.findByIdAndDeletedFalse(1L)).thenReturn(Optional.of(existingCarton));
+        assertThrows(ValidationException.class, () -> cartonService.updateCarton(1L, cartonDTO));
+        verify(cartonRepository, times(1)).findByIdAndDeletedFalse(1L);
+        verifyNoInteractions(modelMapper);
     }
+    @Test
+    public void testUpdateCarton_MissingSize() {
+        CartonDTO cartonDTO = new CartonDTO();
+        cartonDTO.setId(1L);
+        cartonDTO.setName("name");
+        cartonDTO.setAvailableQuantity(10);
+        cartonDTO.setPrice(100.0);
 
+        Carton existingCarton = new Carton();
+        existingCarton.setId(1L);
+        existingCarton.setName("name");
+        existingCarton.setSize("Existing Carton");
+        existingCarton.setAvailableQuantity(5);
+        existingCarton.setPrice(50.0);
+
+        when(cartonRepository.findByIdAndDeletedFalse(1L)).thenReturn(Optional.of(existingCarton));
+        assertThrows(ValidationException.class, () -> cartonService.updateCarton(1L, cartonDTO));
+        verify(cartonRepository, times(1)).findByIdAndDeletedFalse(1L);
+        verifyNoInteractions(modelMapper);
+    }
     @Test
-    void testDeleteCarton_InvalidId() {
-        Long invalidId = 999L;
-        when(cartonRepository.findById(invalidId)).thenReturn(Optional.empty());
-        assertThrows(ApiRequestException.class, () -> cartonService.deleteCarton(invalidId));
-        verify(cartonRepository).findById(invalidId);
-        verify(cartonRepository, never()).delete(any(Carton.class));
+    public void testUpdateCarton_NotAvailableQuantity() {
+        CartonDTO cartonDTO = new CartonDTO();
+        cartonDTO.setId(1L);
+        cartonDTO.setSize("10-19");
+        cartonDTO.setName("name");
+        cartonDTO.setAvailableQuantity(-10);
+        cartonDTO.setPrice(100.0);
+
+        Carton existingCarton = new Carton();
+        existingCarton.setId(1L);
+        existingCarton.setName("name");
+        existingCarton.setSize("Existing Carton");
+        existingCarton.setAvailableQuantity(5);
+        existingCarton.setPrice(50.0);
+
+        when(cartonRepository.findByIdAndDeletedFalse(1L)).thenReturn(Optional.of(existingCarton));
+        assertThrows(ValidationException.class, () -> cartonService.updateCarton(1L, cartonDTO));
+        verify(cartonRepository, times(1)).findByIdAndDeletedFalse(1L);
+        verifyNoInteractions(modelMapper);
     }
+    @Test
```

Comment:
is there a test method that check for successful update of a carton?

Reply:
Yes, we have that test but in other branch which is also in pull-request